client/components/New.js

```diff
@@ -0,0 +1,177 @@
+      {skillCategoryName:'Category Name',skills:['skill-1','skill-2','skill-3'],categoryImg:''},
+      {skillCategoryName:'Category Name',skills:['skill-1','skill-2','skill-3','skill-3','skill-3','skill-3'],categoryImg:''}],
+      roleName:'role Name',lego:'',description:'bla bla bla bla bla bla bla bla bla bla bla bla bla bla ',
+      keywords: [{id:1,text:'keyword-1'},{id:1,text:'keyword-2'},{id:1,text:'keyword-3'}],
+      suggestions:["Banana", "Mango", "Pear", "Apricot"]
+    }
+    this.setRoleProps=this.setRoleProps.bind(this)
+    this.setCategoryDetails=this.setCategoryDetails.bind(this)
+    this.setLego=this.setLego.bind(this)
+    this.create=this.create.bind(this)
+  }
+  setRoleProps(roleName_,keywords_,description_){
+
+    this.setState({
+      roleName: roleName_, keywords: keywords_,description:description_
+    });
+
+  }
+  setLego(legoImg){this.setState({lego:legoImg})}
+
+  setCategoryDetails(id,categoryName,skills,iconUrl){
+    console.log('New.js',iconUrl)
+    let c=this.state.categories
+    c[id-1].skillCategoryName=categoryName;
+    c[id-1].skills=skills;
+    c[id-1].categoryImg=iconUrl;
+    this.setState({
+      categories: c
+    });
+  }
+
+  componentDidUpdate(){console.log(this.state.categories[0].categoryImg)}
+
+  create(){
+      console.log('posting .....')
+      const opts={}
+      opts.name=this.state.roleName
+      opts.description=this.state.description
+      opts.image=this.state.lego
+      opts.keywords=this.state.keywords.map((item,index)=>{return(item.text)})//['a','b','c']
+      opts.skillCategory=[]
+      opts.skillCategory=[1,2,3,4,5,6].map((item,index)=>{
+        return(
+        {         name:this.state.categories[index].skillCategoryName,
+                  skills:this.state.categories[index].skills,
+                  image:this.state.categories[index].categoryImg
+                }
+      )})
+      console.log(opts)
+      const app=this
+      fetch('http://patica-role.mertdogar.com/role', {
+       method: 'post',
+       headers: {
+          'Accept': 'application/json, text/plain, */*',
+          'Content-Type': 'application/json'
+        },
+       body: JSON.stringify(opts)
+           }).then(function(response) {
+             return response.json();
+           }).then(function(data) {
+              app.setState({ error: data.error.message });
+
+     });
+  }
+  render() {
+
+    return (
+      <div className='col' >
+          <div className='row' style={{justifyContent:'center'}}>
+           <div className='' >
+              <RoleLabel setRoleProps={this.setRoleProps} roleName={this.state.roleName} keywords={this.state.keywords}
+                description={this.state.description} suggestions={this.state.suggestions}
+              />
+          </div>
+         </div>
+          <div className='main-container' >
+             <div className="col-left" >
+               <div className="item" >
+               <SkillCategoryLabel
```

Comment:
Again, SkillCategoryLabel must be rendered dynamically using Array.map or a similar approach. What happens if Niels wants to have 5 or 8 skill categories?